lib/common/socketpool.c

```diff
@@ -406,30 +443,49 @@ void h2o_socketpool_connect(h2o_socketpool_connect_request_t **_req, h2o_socketp
     if (_req != NULL)
         *_req = NULL;
 
+    size_t target = SIZE_MAX;
+    h2o_linklist_t *sockets = NULL;
+
     /* fetch an entry and return it */
     pthread_mutex_lock(&pool->_shared.mutex);
     destroy_expired(pool);
-    while (1) {
-        if (h2o_linklist_is_empty(&pool->_shared.sockets))
-            break;
-        entry = H2O_STRUCT_FROM_MEMBER(struct pool_entry_t, link, pool->_shared.sockets.next);
-        h2o_linklist_unlink(&entry->link);
+
+    /* TODO is this needed in this critical section? */
```

Comment:
No. I think that you should move the call to `lookup_target` outside of the critical section.